yb-voyager/src/srcdb/postgres.go

```diff
@@ -1502,3 +1509,26 @@ func (pg *PostgreSQL) listSchemasMissingUsagePermission() ([]string, error) {
 
 	return schemasMissingUsagePermission, nil
 }
+
+func (pg *PostgreSQL) CheckDependencies() error {
+	var errs []string
+
+	// List of pg binaries to check
+	binaries := []string{"pg_dump", "pg_restore"}
+	sourceDBVersion := pg.GetVersion()
+
+	for _, binary := range binaries {
+		_, err := GetAbsPathAndCheckVersionOfPGCommand(binary, sourceDBVersion)
+		if err != nil {
+			errs = append(errs, err.Error())
+		} else {
+			log.Infof("%s is compatible", binary)
+		}
+	}
+
+	if len(errs) > 0 {
+		return errors.New(strings.Join(errs, "\n"))
+	}
+
```

Comment:
how is the output coming out for this.
I am just thinking if that is a good way to combine all the errors and make New().

how about simply returning
```
    if len(errs) > 0 {
        return fmt.Errorf("%w or %v", errs)
    }
```

Reply:
Done